temboardui/plugins/monitoring/handlers/alerting.py

```diff
@@ -0,0 +1,319 @@
+
+            # Find host_id & instance_id
+            host_id = get_host_id(self.db_session, instance.hostname)
+            instance_id = get_instance_id(self.db_session, host_id,
+                                          instance.pg_port)
+            query = self.db_session.query(
+                        Check.name, CheckState.key, CheckState.state
+                    ).filter(
+                        Check.host_id == host_id,
+                        Check.instance_id == instance_id,
+                        Check.check_id == CheckState.check_id
+                    ).order_by(
+                        Check.name,
+                        CheckState.key
+                    )
+            data = [{'name': r.name, 'key': r.key, 'state': r.state}
+                    for r in query]
+            self.db_session.close()
+            return JSONAsyncResult(http_code=200, data=data)
+
+        except (TemboardUIError, Exception) as e:
+            self.logger.exception(e)
+            try:
+                self.db_session.close()
+            except Exception:
+                pass
+            if (isinstance(e, TemboardUIError)):
+                return JSONAsyncResult(http_code=e.code,
+                                       data={'error': e.message})
+            else:
+                return JSONAsyncResult(http_code=500,
+                                       data={'error': e.message})
+
+    @tornado.web.asynchronous
+    def get(self, agent_address, agent_port):
+        run_background(self.get_check_states, self.async_callback,
+                       (agent_address, agent_port))
+
+
+class AlertingDataCheckResultsHandler(CsvHandler):
+
+    def get_data_check_results(self, address, port, check_name):
+        try:
+            instance = None
+            role = None
+            no_error = 0
+
+            self.load_auth_cookie()
+            self.start_db_session()
+
+            role = self.current_user
+            if not role:
+                raise TemboardUIError(302, "Current role unknown.")
+
+            instance = get_instance(self.db_session, address, port)
+            if not instance:
+                raise TemboardUIError(404, "Instance not found.")
+            if 'monitoring' not in [plugin.plugin_name
+                                    for plugin in instance.plugins]:
+                raise TemboardUIError(408, "Plugin not active.")
+
+            # Find host_id & instance_id
+            host_id = get_host_id(self.db_session, instance.hostname)
+            instance_id = get_instance_id(self.db_session, host_id,
+                                          instance.pg_port)
+
+            self.db_session.expunge_all()
+
+            # Arguments
+            start = self.get_argument('start', default=None)
+            end = self.get_argument('end', default=None)
+            key = self.get_argument('key', default=None)
+            # Return 200 with empty list when an error occurs
```

Comment:
I don't understand this comment.

Reply:
If `no_error=1` then we do not return HTTP error if something goes wrong.